app/script/view_model/ConversationInputViewModel.coffee

```diff
@@ -132,11 +132,48 @@ class z.ViewModel.ConversationInputViewModel
       @logger.log "Ephemeral timer for conversation '#{@conversation_et().display_name()}' is now at '#{@conversation_et().ephemeral_timer().toString()}'."
 
   upload_images: (images) =>
+    if @_is_hitting_upload_limit images
+      return
+
     for image in images
       return @_show_upload_warning image if image.size > z.config.MAXIMUM_IMAGE_FILE_SIZE
 
     @conversation_repository.upload_images @conversation_et(), images
 
+  upload_files: (files) =>
+    if @_is_hitting_upload_limit files
+      return
+
+    for file in files
+      if file.size > z.config.MAXIMUM_ASSET_FILE_SIZE
+        amplify.publish z.event.WebApp.ANALYTICS.EVENT, z.tracking.EventName.FILE.UPLOAD_TOO_BIG,
+          {size: file.size, type: file.type}
+        amplify.publish z.event.WebApp.AUDIO.PLAY, z.audio.AudioType.ALERT
+        window.setTimeout ->
+          amplify.publish z.event.WebApp.WARNING.MODAL, z.ViewModel.ModalType.UPLOAD_TOO_LARGE,
+            data: z.util.format_bytes(z.config.MAXIMUM_ASSET_FILE_SIZE)
+        , 200
+        return
+
+    @conversation_repository.upload_files @conversation_et(), files
+
+  on_drop_files: (dropped_files) =>
+    images = []
+    files = []
+
+    if @_is_hitting_upload_limit dropped_files
+      return
+
+    for file in dropped_files
+      switch
+        when file.type in ['.jpg-large', 'image/jpeg', 'image/png', 'image/bmp', 'image/gif']
```

Comment:
In our `config.coffee` we have `SUPPORTED_IMAGE_TYPES`. What about adding `.jpg-large` there and reusing `SUPPORTED_IMAGE_TYPES`? See: https://github.com/wireapp/wire-webapp/blob/dev/app/script/config.coffee#L77